analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -32,10 +34,35 @@ internal static class Binary
                 ? context.SetOperationConstraint(newConstraint)
                 : context.State;
 
-        public static ProgramState LearnBranchingConstraint(ProgramState state, IBinaryOperationWrapper binary, bool useOpposite) =>
-            binary.OperatorKind.IsAnyEquality()
+        public static ProgramState LearnBranchingConstraint(ProgramState state, IBinaryOperationWrapper binary, bool useOpposite)
+        {
+            state = binary.OperatorKind.IsAnyEquality()
                 ? LearnBranchingConstraint<ObjectConstraint>(state, binary, useOpposite) ?? LearnBranchingConstraint<BoolConstraint>(state, binary, useOpposite) ?? state
                 : state;
+            state = LearnNullableValueTypeConstraint(state, binary, useOpposite);
+            return state;
+        }
+
+        private static ProgramState LearnNullableValueTypeConstraint(ProgramState state, IBinaryOperationWrapper binary, bool useOpposite)
+        {
+            var nullableOperation = BranchPredicate(binary, (side, opposite) => side.Type.IsNullableValueType() && opposite.Type.IsValueType);
+            if (nullableOperation is not null && nullableOperation.ResolveTrackedSymbol(state) is { } symbol)
```

Comment:
`ResolveTrackedSymbol` is basically `TrackedSymbol` but goes back to a "root", so for an expression `arg?.Length`, it returns `arg`. Unfortunately, some unresolvable flow capture references (captureId is not in the `CaptureOperation` dictionary) prevent that.

Reply:
If we can not resolve the capture, we may consider doing the following:
* SetOperationConstraint for NotNull on the left side of the operation (`if (arg?.ToString()?.Length > 3)`) in the `!useOpposite` case.
* Inherit the constraint in the following operation kinds:
  * [x] Conversion
  * [ ] FlowCapture
  * [ ] PropertyReference
  * [ ] MethodReference
  * [ ] Other references
This may break all kinds of other cases, though.